File: 2021-rabbitmq-jpa/domain/jpa/src/main/code/sample/domain/jpa/repository/CommentRepository.java
package sample.domain.repository;

import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;
import sample.domain.model.ArticleEntity;
import sample.domain.model.CommentEntity;

import java.util.Optional;

@Repository
public interface CommentRepository extends JpaRepository<CommentEntity, Long> {
    Optional<CommentEntity> findFirstByArticleOrderByIdDesc(ArticleEntity article);
}

File: 2022-declarative/domain/board/src/main/kotlin/org/scriptonbasestar/domain/board/persistence/CommentRepository.kt
package org.scriptonbasestar.domain.board.persistence

import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.jpa.repository.JpaSpecificationExecutor
import org.springframework.stereotype.Repository
import java.util.*

@Repository
interface CommentRepository :
    JpaRepository<CommentEntity, Long>,
    JpaSpecificationExecutor<CommentEntity> {
    fun findAllByArticle(article: ArticleEntity, pageable: Pageable): Page<CommentEntity>
    fun findOneByArticleAndUuid(article: ArticleEntity, uuid: UUID): Optional<CommentEntity>
}
